### PR TITLE
Allow SSpec files as Scarpe tests

### PR DESCRIPTION
**Pull request by noahgibbs**
### Description

It turns out there was a bug in the file loader that was doing something I didn't want. This fixes that, adds tests and adds test_helper code so you can add Shoes-Spec sspec files as tests in Scarpe.

### Checklist

- [ ] Run tests locally

### CODE REVIEW

File: test/test_sspec_infrastructure.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      Shoes.app do
+      end
+      ----------- test code
+      skip
+    SSPEC
+  end
+
+  def test_assertion_fail
+    run_scarpe_sspec_code(<<~'SSPEC', expect_result: :fail)
+      ---
```
Comment (Schwad) [inline test/test_sspec_infrastructure.rb:56]: How should I implement this in the wild?

Define it as an .sspec file under the test directory?

Or am I able to write my examples/ code in this manner and it'll get pulled in for tests?

or do I need to run it in the "run_scarpe_sspec_code" block, as a ShoesSpecLoggedCase, in the test directory?

**Review by Schwad — APPROVED** [review]

File: test/test_sspec_infrastructure.rb (hunk 1 above)
Comment (noahgibbs) [reply test/test_sspec_infrastructure.rb:56]: Right now you'll need to run_scarpe_sspec. So we'll need an equivalent of test/test_examples.rb, where we go through a directory and call run_scarpe_sspec on each of them. Right now you'll *also* need to specify whether you expect the test to fail -- if something gets an error or a skip and you expect that, you'll need to tell it so. Eventually we should have frontmatter to specify that...

For Shoes-Spec I'm planning to add a rake task to do it. That'll make it easier to clone a Shoes-Spec dir somewhere and then run through it without requiring that the Shoes-Spec dir always be in the same place in the file system.

Comment (Schwad) [reply test/test_sspec_infrastructure.rb:56]: > Right now you'll need to run_scarpe_sspec.

That's totally fine with me. More than happy to run that in an sspec block!

Comment (Schwad) [reply test/test_sspec_infrastructure.rb:56]: It should catch on `rake test` if I want to add some?

Comment (noahgibbs) [reply test/test_sspec_infrastructure.rb:56]: Again, you'd need a test_*.rb file to run it. But you could easily call run_scarpe_sspec on the file. As with test_examples, the best way to do a directory would be to list the files in there and make one Minitest test for each one. Something like:

```
class TestShoesSpecFiles < ShoesSpecLoggedTest
  self.logger_dir = File.expand_path("#{__dir__}/../logger")

  sspecs_to_test = Dir["spec/**/*.sspec"]
  sspecs_to_test.each.with_index do |filename, n|
    define_method("test_sspec_#{n}") do
      run_scarpe_sspec(filename)
    end
  end
end
```

I'm sure you could do better than that, but I'm just typing into a comment right now :-)

Comment (noahgibbs) [reply test/test_sspec_infrastructure.rb:56]: If I do a rake task to run the Shoes-Spec files I'll do something similar -- find all the specs, find the results files, and write a Minitest test that expects it to pass, fail, error, skip, etc. according to what's in the results YAML. A little more complicated, but not ridiculously so.

**Pull request merged**